From 855120109fbfca89477c25ad75004cb7ff0b4170 Mon Sep 17 00:00:00 2001
Message-ID: <855120109fbfca89477c25ad75004cb7ff0b4170.1706151403.git.sharlatanus@gmail.com>
From: Sharlatan Hellseher <sharlatanus@gmail.com>
Date: Thu, 25 Jan 2024 02:48:04 +0000
Subject: [PATCH] gnu: python-pyro-ppl: Fix build.

As seen in https://ci.guix.gnu.org/build/3347962/log/raw.

* gnu/packages/machine-learning.scm (python-pyro-ppl): Fix build by
disable more failing tests.
[build-system]: Swap to pyproject-build-system.
[arguments]: Refactor to use <#:test-flags> instead of custom 'check
phase. Add "-n" flag to pytest to accelerate tests. Silent more failing
tests.

Change-Id: Ia1069dee9e7e9534f694e01a847ae27d7a787391
---
 gnu/packages/machine-learning.scm | 37 +++++++++++++++++--------------
 1 file changed, 20 insertions(+), 17 deletions(-)

diff --git a/gnu/packages/machine-learning.scm b/gnu/packages/machine-learning.scm
index 5ae210f375..4757cd7265 100644
--- a/gnu/packages/machine-learning.scm
+++ b/gnu/packages/machine-learning.scm
@@ -22,6 +22,7 @@
 ;;; Copyright © 2023 Navid Afkhami <navid.afkhami@mdc-berlin.de>
 ;;; Copyright © 2023 Zheng Junjie <873216071@qq.com>
 ;;; Copyright © 2023 Troy Figiel <troy@troyfigiel.com>
+;;; Copyright © 2043 Sharlatan Hellseher <sharlatanus@gmail.com>
 ;;;
 ;;; This file is part of GNU Guix.
 ;;;
@@ -4848,24 +4849,26 @@ (define-public python-pyro-ppl
        (file-name (git-file-name name version))
        (sha256
         (base32 "0n1vsih99pvswcaygdxkc6kq6r48ny130z6ca8pp3281396r2ykw"))))
-    (build-system python-build-system)
+    (build-system pyproject-build-system)
     (arguments
-     `(#:phases
-       (modify-phases %standard-phases
-         (replace 'check
-           (lambda* (#:key tests? #:allow-other-keys)
-             ;; This tests features that are only implemented when non-free
-             ;; software is available (Intel MKL or CUDA).
-             (for-each delete-file
-                       (list "tests/distributions/test_spanning_tree.py"
-                             "tests/infer/mcmc/test_mcmc_api.py"))
-
-             ;; Four test_gamma_elbo tests fail with bad values for unknown
-             ;; reasons.
-             (delete-file "tests/distributions/test_rejector.py")
-             ;; This test fails sometimes.
-             (delete-file "tests/optim/test_optim.py")
-             (invoke "pytest" "-vv" "--stage=unit"))))))
+     (list
+      ;; Tests take about 45min to pass.
+      #:test-flags
+      #~(list
+         "--stage=unit"
+         "-n" (number->string (parallel-job-count))
+         ;; This tests features that are only implemented when non-free
+         ;; software is available (Intel MKL or CUDA).
+         "--ignore=tests/distributions/test_spanning_tree.py"
+         "--ignore=tests/infer/mcmc/test_mcmc_api.py"
+         ;; Four test_gamma_elbo tests fail with bad values for unknown
+         ;; reasons.
+         "--ignore=tests/distributions/test_rejector.py"
+         ;; This test fails sometimes.
+         "--ignore=tests/optim/test_optim.py"
+         ;; Failing scipy: TypeError: unsupported operand type(s) for -:
+         ;; 'function' and 'int'.
+         "--ignore=tests/distributions/test_stable.py")))
     (propagated-inputs
      (list python-numpy
            python-opt-einsum

base-commit: ffa73f8ef2a6cb2b93deb38b1c88a94707e7d907
-- 
2.41.0

